### PR TITLE
404369 support multiple tabs

### PR DESCRIPTION
**Pull request by feedmypixel**
- Support multiple tabs on the deploy flow
- Create first cut of a common `multistep-form` plugin
- Put `pulse` behind a flag. Its getting super hard to work with in `dev`

## Demo
multiple-tabs

Screenshot 2024-08-19 at 18 25 56
Screenshot 2024-08-19 at 18 26 06
Screenshot 2024-08-19 at 18 26 14

### CODE REVIEW

File: src/server/common/helpers/start-server.js [hunk 1]
```diff
@@ -1,23 +1,16 @@
...
-  } catch (error) {
-    const logger = createLogger()
-    logger.info('Server failed to start :(')
-    logger.error(error)
-  }
+  server.logger.info('Server started successfully')
+  server.logger.info(
+    `Access your frontend on http://localhost:${config.get('port')}`
+  )
+  return server
```
Comment (christopherjturner) [inline src/server/common/helpers/start-server.js:13]: was the try/catch redundant or removed as part of debugging the dev/restart issue?

File: src/config/config.js [hunk 2]
```diff
@@ -136,25 +140,23 @@ const config = convict({
...
     doc: 'If this application running in the production environment',
     format: Boolean,
-    default: process.env.NODE_ENV === 'production'
+    default: isProduction
   },
   isDevelopment: {
     doc: 'If this application running in the development environment',
     format: Boolean,
-    default: process.env.NODE_ENV !== 'production'
+    default: !isProduction
```
Comment (chotai) [inline src/config/config.js:148]: if NODE_ENV === 'test' then `isDevelopment` will also be true - I know you haven't changed the logic here

**Review by chotai — APPROVED** [review]
I liked `availableMemoryOptions` not being in formValues but different preferences I suppose

File: src/config/config.js (hunk 2 above)
Comment (feedmypixel) [reply src/config/config.js:148]: yeah theres a few other things incorrectly hooked into this. I'll have a look. At the time it blew up and so I kept it the same

**Comment by feedmypixel** [comment]
> I liked `availableMemoryOptions` not being in formValues but different preferences I suppose

Yeah this was part of making the work generic. Brining all form things into `formValues` gives a holder for everything to be referenced from. Making this generic work easier to adopt when you need to pass form control options around

File: src/server/common/helpers/start-server.js (hunk 1 above)
Comment (feedmypixel) [reply src/server/common/helpers/start-server.js:13]: This work has been moved to the catch in the added to the `startServer.catch()` call - this for the moment seems to of fixed the issue

Comment (feedmypixel) [reply src/server/common/helpers/start-server.js:13]: When we move to `ESM` I'll replace this with a high level `await`, same same as the templates

**Pull request merged**